### DISCUSSION TITLE
Failure modes of Sync Webhooks / Communication between Saleor and Webhook Handlers

### DISCUSSION THREAD

**Discussion by rrrnld** [saleor/saleor#10890] [Q&A] [answered]
Hi,

is there any way to mark a webhook response as required? My tests suggest that if there's an internal error in a handler that generates a response, the default mode for saleor is to just ignore it and carry on? In the case of the [tax webhooks](https://docs.saleor.io/docs/3.x/developer/extending/apps/synchronous-webhooks/tax-webhooks) that would mean tax calculation is simply skipped, which is suboptimal (and different to a failure inside a plugin doing the same).

Are my observations correct? If yes, is there any way to change this behavior, or am I maybe missing something?

To expand this question: Do we have any way to communicate back to Saleor from within a webhook handler outside of the expected response format? Is there a way to set metadata from within a synchronous webhook response?

**Comment by patrys** [comment, marked as answer]
Hello, @heyarne. This behavior is by design. Saleor operates under the principle of not allowing component failures to propagate. If an App is down, Saleor will allow the order to be placed, assuming additional work for your customer support team is preferable to losing sales.
